conveyor/worker.go

```diff
@@ -368,13 +412,15 @@ func (w *worker) processSignalsSuspending(elements []queue.OutputElement) int {
 			numSignals++
 			switch el.GetItemType() {
 			case PendingPulseSignal:
-				log.Warn("[ processSignalsSuspending ] Must not be PendingPulseSignal here. Skip it")
+				w.ctxLogger.Warn("[ processSignalsSuspending ] Got PendingPulseSignal. Must not be PendingPulseSignal here. Skip it")
 			case ActivatePulseSignal:
+				w.ctxLogger.Info("[ processSignalsSuspending ] Got ActivatePulseSignal. Set slot state to 'Initializing'")
 				w.changePulseState()
-				w.slot.slotState = Initializing
+				w.changeSlotState(Initializing)
 				w.activatePulseSync = el.GetData().(queue.SyncDone)
-				log.Info("[ processSignalsSuspending ] Set slot state to 'Initializing'")
 			case CancelSignal:
+				w.ctxLogger.Info("[ processSignalsSuspending ] Got CancelSignal. Set slot state to 'Initializing")
+				w.changeSlotState(Initializing)
```

Comment:
Why change state to `Initializing` after cancel signal?

Reply:
I will add one more state, and change to it. It will be more clear